src/pages/Auth/AuthPage.tsx

```diff
@@ -0,0 +1,171 @@
+
+const Link = styled(NavLink)({
+  color: grey[700],
+  textDecoration: 'none',
+  '&:hover': {
+    color: grey[500],
+  },
+});
+
+const InputField = styled(TextField)({
+  '& label.Mui-focused': {
+    color: indigo[700],
+  },
+  '& label.Mui-error': {
+    color: red[700],
+  },
+  '& .MuiOutlinedInput-root': {
+    '&.Mui-focused fieldset': {
+      borderColor: indigo[700],
+      borderWidth: '1px',
+    },
+    '&.Mui-error fieldset': {
+      borderColor: red[700],
+      borderWidth: '1px',
+    },
+  },
+});
+
+const AuthPage: FC = () => {
+  const localization = useTranslation();
+  const location = useLocation();
+  const isLogin = location.pathname === '/login';
+  const showMsg = useAlert();
+  const [isLoading, setIsLoading] = useState(false);
+  const navigate = useNavigate();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<IFormData>({
+    mode: 'onSubmit',
+    reValidateMode: 'onSubmit',
+  });
+
+  const onSubmit: SubmitHandler<IFormData> = async (data: IFormData) => {
+    if (isLogin) {
+      try {
+        setIsLoading(true);
+        await signInWithEmailAndPassword(authState, data.email, data.password);
+        showMsg({ type: 'success', content: localization.auth.loginSuccess });
+        navigate('/main');
+      } catch (e) {
+        if (e instanceof FirebaseError)
+          return showMsg({ type: 'error', content: fBaseErrReader(e.message) });
+      } finally {
+        setIsLoading(false);
+      }
+    } else {
+      try {
+        setIsLoading(true);
+        await createUserWithEmailAndPassword(authState, data.email, data.password);
+        showMsg({ type: 'success', content: localization.auth.accountCreated });
+      } catch (e) {
+        if (e instanceof FirebaseError)
+          return showMsg({ type: 'error', content: fBaseErrReader(e.message) });
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    reset();
+  };
+
+  return (
+    <Grid container direction="column" alignItems="center" justifyContent="center">
+      <CssBaseline />
+      <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square borderRadius={2}>
+        <PaperCss>
+          <Typography component="h1" variant="h5">
+            {isLogin ? localization.auth.login : localization.auth.signup}
+          </Typography>
+          <Form onSubmit={handleSubmit(onSubmit)}>
+            <InputField
+              variant="outlined"
+              margin="normal"
+              required
+              fullWidth
+              id="email"
+              label="Email"
+              autoComplete="email"
+              autoFocus
+              error={!!errors.email}
+              helperText={errors.email ? errors.email.message : ''}
+              {...register('email', {
+                required: true,
+                pattern: {
+                  value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
+                  message: 'invalid email address',
```

Comment:
the message is not localized
```suggestion
                  message: localization.auth.emailMessage,
```